tests/test_wallet.py

```diff
@@ -568,7 +568,7 @@ def test_transaction_annotations(node_factory, bitcoind):
 
 @unittest.skipIf(VALGRIND, "It does not play well with prompt and key derivation.")
 def test_hsm_secret_encryption(node_factory):
-    l1 = node_factory.get_node()
+    l1 = node_factory.get_node(may_fail=True)  # May fail when started without key
```

Comment:
Hmm it shouldn't, should it ? At this point it doesn't need a key?

Reply:
The `may_fail` argument basically is just there for the teardown checks which look into the node and check if it failed the last time in the test.

You are right that with the test as it is right now (last variant exits correctly) we should not need this, let me try if it passes correctly without.